hm-module.nix

```diff
@@ -0,0 +1,57 @@
+self:
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+with lib;
+
+let
+  cfg = config.programs.krewfile;
+  finalPackage = self.packages.${pkgs.system}.krewfile;
+  krewfileContent = pkgs.writeText "krewfile" (concatStringsSep "\n" cfg.plugins);
+  args = if cfg.upgrade then "-upgrade" else "";
+in
+{
+  options.programs.krewfile = {
+    enable = mkEnableOption "krewfile";
+
+    krewPackage = mkOption {
+      type = types.package;
+      default = pkgs.krew;
+      defaultText = literalExpression "pkgs.krew";
+      description = "Krew package to install.";
+    };
+
+    plugins = mkOption {
+      type = with types; listOf str;
+      default = [ ];
+      defaultText = literalExpression "[ "edit-status" ]";
+      description = "List of plugins to be installed.";
+    };
+
+    upgrade = mkOption {
+      type = types.bool;
+      default = false;
+      defaultText = literalExpression "false";
+      description = "Enable auto update of plugins.";
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    home.packages = [ finalPackage cfg.krewPackage ];
+
+    home.sessionVariables.PATH = "$HOME/.krew/bin:$PATH";
+
+    home.activation.krew = hm.dag.entryAfter [ "writeBoundary" ] ''
+      PATH="$HOME/.krew/bin:$PATH"
+      run ${cfg.krewPackage}/bin/${cfg.krewPackage.pname} update
```

Comment:
I think this can be removed it's already implemented in krewfile itself 🙂 
Ref: https://github.com/brumhard/krewfile/blob/02d04f38ea0d1b8de4da3b29bc861f8883c9b9e9/main.go#L73

```suggestion
```